### PR TITLE
Reduced Time Complexity to O(sqrt(n))

### PR DESCRIPTION
**Pull request by Hannibal404**
### Describe your change:

* [x] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/factors.py [hunk 1]
```diff
@@ -1,3 +1,6 @@
+from math import sqrt
+
```
Comment (Abinashbunty) [inline maths/factors.py:2]: import doctest and call the `testmod()`

**Review by Abinashbunty — CHANGES_REQUESTED** [review]

File: maths/factors.py [hunk 2]
```diff
@@ -9,7 +12,21 @@ def factors_of_a_number(num: int) -> list:
...
+    if num == 1:
+        return facs
+    facs.append(num)
+    for i in range(2, int(sqrt(num)) + 1):
+        if num % i == 0:  # If i is a factor of num
+            facs.append(i)
+            d = num // i  # num//i is the other factor of num
+            if d != i:  # If d and i are distinct
+                facs.append(d)  # we have found another factor
+    facs.sort()
```
Comment (CaedenPH) [inline maths/factors.py:28]: The `sort` function means that the time complexity becomes `O(n sqrt(n))`

File: maths/factors.py (hunk 1 above)
Comment (Hannibal404) [reply maths/factors.py:3]: Done

**Review by Abinashbunty — APPROVED** [review]

File: maths/factors.py (hunk 2 above)
Comment (CaedenPH) [reply maths/factors.py:29]: > It would actually be O(sqrt(n) * log(sqrt(n))) since the length of the list is <= 2*sqrt(n)

I see

Comment (Hannibal404) [reply maths/factors.py:29]: @CaedenPH requesting you to please approve the review so the PR may be merged. Thanks

**Review by CaedenPH — APPROVED** [review]

**Review by prateekiiest — APPROVED** [review]

**Pull request merged**